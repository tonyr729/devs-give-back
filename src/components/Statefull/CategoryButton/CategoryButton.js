import React, { Component } from 'react';

class CategoryButton extends Component {
  constructor() {
    super();
    this.state = {
      selected: false
    };
  }

  handleClick = () => {
    this.props.selectCategory(this.props.category);
    this.setState({
      selected: !this.state.selected
    });
  }

  render() {
    let cardClass = this.state.selected ? "selected-category" : "category-button";
    return (
      <button onClick={this.handleClick} className={cardClass}>{this.props.category}</button>
    );
  }
}

export default CategoryButton;