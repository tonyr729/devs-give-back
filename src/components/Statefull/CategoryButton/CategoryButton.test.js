import React from 'react';
import { shallow } from 'enzyme';
import CategoryButton from './CategoryButton';

describe('CategoryButton', () => {
  let wrapper;
  let mockSelectCategory;

  beforeEach(() => {
    mockSelectCategory = jest.fn();
    wrapper = shallow(<CategoryButton selectCategory={mockSelectCategory} category={'category'}  />);
  })

  it('should match snapshot', () => {

    expect(wrapper).toMatchSnapshot();

  });
  
  describe('handleClick', () => {
    it('should call prop methods and values', () => {
      wrapper.instance().handleClick()

      expect(mockSelectCategory).toHaveBeenCalled();
    });
  });
});