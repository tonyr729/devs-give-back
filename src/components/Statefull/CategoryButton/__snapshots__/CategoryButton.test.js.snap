// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryButton should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CategoryButton
    category="category"
    selectCategory={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "category",
      "className": "category-button",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "category",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "category",
        "className": "category-button",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "category",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
